q.js

```diff
@@ -1012,6 +1027,14 @@ function trackRejection(promise, reason) {
     if (!trackUnhandledRejections) {
         return;
     }
+    if(typeof process === "object" && typeof process.emit === "function"){
+        Q.nextTick.runAfter(function(){
+            if(array_indexOf(unhandledRejections, promise) !== -1){
+                process.emit("unhandledRejection", reason, promise);
```

Comment:
I would prefer to just have optional Q method hooks for browser support, like Q.onerror, or support a delegate object with method hooks.